docs/source/developer/installing.rst

```diff
@@ -121,6 +146,49 @@ with the arguments of your choosing, e.g.:
 
     (py37) $ pytest tests/test_s3.py -x -k replication
 
-Note you'll have to environment variables setup appropriately per the tox.ini
-for provider credentials.
+Note that you'll have to set up environment variables appropriately per the tox.ini
+for provider credentials. See below for the best way to do that.
+
+
+Installing in Your Own Virtual Environment
+------------------------------------------
+
+Running directly from a tox sandbox, while very easy to set up, might not be
+the most comfortable way of working. You might want to create your own virtual
+environment and use that for running Custodian. This can be done using the ``venv``
+module. It can be done right inside the cloned Cloud Custodian repository:
+
+.. code-block:: bash
+
+    $ python3 -m venv .
+
+The above command assumes the current directory is the Cloud Custodian checkout.
+
+Next, you'll need to install all the development dependencies. Cloud Custodian uses
+`poetry <https://python-poetry.org>`_ for packaging and dependency management.
+Since Cloud Custodian needs to install several sub-dependencies, poetry has to be
+installed first, to make sure they all share the same venv:
+
+.. code-block:: bash
+
+    $ source bin/activate
+    (cloud-custodian) $ pip install poetry
```

Comment:
we should reference upstream recommends re install versus pip install on poetry.. aka poetry is a pip replacement. it can work in this mode, but that is not per upstream recommends or best practices.

Reply:
Replied above. This won't be necessary. We can follow best practices after all.